marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java

```diff
@@ -343,7 +357,97 @@ public void upgradeProject() throws IOException {
     @Override public String getUserModulesDeployTimestampFile() {
         return Paths.get(projectDirString, ".tmp", USER_MODULES_DEPLOY_TIMESTAMPS_PROPERTIES).toString();
     }
-
+    
+    //copying only the required old hub-internal-config db and server files to new locations
+    //the security files (users, roles, privileges etc are not copied from old hub-internal-config)
+    private void upgradeHubInternalConfig(Path sourceDir, Path renamedSourceDir, Set<String> hubConfigFiles) throws IOException {
+        if (Files.exists(sourceDir)) {
+            Files.walk(Paths.get(sourceDir.toUri()))
+            .filter( f -> !Files.isDirectory(f))
+            .forEach(f -> {
+                String path = f.toFile().getName();
+                //copy jobs,staging db to hub-internal-config
+                if(path.toLowerCase().equals("job-database.json")
+                || path.toLowerCase().equals("staging-database.json")) {
+                    try {
+                        FileUtils.copyFile(f.toFile(), getHubDatabaseDir().resolve(f.getFileName()).toFile());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+              //copy modules,final db to ml-config
+                if(path.toLowerCase().equals("modules-database.json")
```

Comment:
Is there a chance that these files already existing in ml-config? In which case, we should have a check saying - if the file exists, then don't replace it? I don't know what's best.

Reply:
During upgrade process, we first call initProject(). So "modules-database.json" file will be present in ml-config but we are overwriting it with whatever user had in 3.0. The same is true for the other db and server files